### PR TITLE
nancy lint fixes

### PR DESCRIPTION
**Pull request by Bhagyashreek8**

### CODE REVIEW

File: cmd/provisioner/main.go [hunk 1]
```diff
@@ -70,11 +70,11 @@ var leaseRetryPeriod = flag.Duration(
 	"How often lease acquisition and renewal should be retried",
 )
 
-var leaseTermLimit = flag.Duration(
-	"leaseTermLimit",
-	10*time.Minute,
-	"Maximum time that a provisioner can maintain a lease",
-)
+//var leaseTermLimit = flag.Duration(
```
Comment (ambiknai) [inline cmd/provisioner/main.go:73]: Bhagya is this unused declaration? if so can this be removed.

File: driver/driver.go [hunk 2]
```diff
@@ -50,14 +50,14 @@ const (
...
-	removeAll          = os.RemoveAll
-	hostname, anyerror = os.Hostname()
+	command   = exec.Command
+	stat      = os.Stat
+	unmount   = syscall.Unmount
+	mount     = syscall.Mount
+	writeFile = ioutil.WriteFile
+	mkdirAll  = os.MkdirAll
+	removeAll = os.RemoveAll
+	//hostname, anyerror = os.Hostname()
```
Comment (ambiknai) [inline driver/driver.go:60]: same here. Can we remove if commented out lines are unused?

File: provisioner/ibm-s3fs-provisioner_test.go [hunk 3]
```diff
@@ -103,17 +103,17 @@ const (
...
+	optionParallelCount      = "parallel-count"
+	optionMultiReqMax        = "multireq-max"
+	optionStatCacheSize      = "stat-cache-size"
+	optionS3FSFUSERetryCount = "s3fs-fuse-retry-count"
+	optionTLSCipherSuite     = "tls-cipher-suite"
+	optionDebugLevel         = "debug-level"
+	optionCurlDebug          = "curl-debug"
+	optionKernelCache        = "kernel-cache"
+	optionOSEndpoint         = "object-store-endpoint"
+	//optionRegion                  = "region"
```
Comment (ambiknai) [inline provisioner/ibm-s3fs-provisioner_test.go:116]: same here.. can we remove?

**Review by ambiknai — CHANGES_REQUESTED** [review]
Bhagya.. I have added minor comments.. As I see changes in scripts used to build driver [internal repo ] , can we test it once by building image.

File: cmd/provisioner/main.go (hunk 1 above)
Comment (Bhagyashreek8) [reply cmd/provisioner/main.go:73]: yes this is unused declaration. Verified it. kept it commented just incase if this is required in scenarios.

File: driver/driver.go (hunk 2 above)
Comment (Bhagyashreek8) [reply driver/driver.go:60]: kept it commented for reference if required

File: provisioner/ibm-s3fs-provisioner_test.go (hunk 3 above)
Comment (Bhagyashreek8) [reply provisioner/ibm-s3fs-provisioner_test.go:116]: kept it commented for reference if required

**Comment by Bhagyashreek8** [comment]
I have tested by building the images locally, had to make some tweaks to build locally, verified functionality works fine.

File: provisioner/ibm-s3fs-provisioner.go [hunk 4]
```diff
@@ -137,7 +136,7 @@ var _ controller.Provisioner = &IBMS3fsProvisioner{}
 var writeFile = ioutil.WriteFile
 
 func UnixConnect(addr string, t time.Duration) (net.Conn, error) {
-	unix_addr, err := net.ResolveUnixAddr("unix", addr)
+	unix_addr, err := net.ResolveUnixAddr("unix", addr) // nolint:ineffassign
```
Comment (mssachan) [inline provisioner/ibm-s3fs-provisioner.go:139]: Maybe replacing `unix_addr, err := net.ResolveUnixAddr("unix", addr)` ---> `unix_addr, _ := net.ResolveUnixAddr("unix", addr)` should fix this as we are not utilizing this `err` futher

Also is there any possibility that `net.ResolveUnixAddr` might fail? If yes, then I think we  may need to return `err` after this call itself.

File: utils/grpc-client/grpc_client_test.go [hunk 5]
```diff
@@ -98,7 +98,7 @@ func Test_GrpcDial_Error(t *testing.T) {
 }
 
 func UnixConnect(addr string, t time.Duration) (net.Conn, error) {
-	unix_addr, err := net.ResolveUnixAddr("unix", addr)
+	unix_addr, err := net.ResolveUnixAddr("unix", addr) // nolint:ineffassign
```
Comment (mssachan) [inline utils/grpc-client/grpc_client_test.go:101]: Same here

**Review by mssachan — CHANGES_REQUESTED** [review]

File: provisioner/ibm-s3fs-provisioner.go (hunk 4 above)
Comment (Bhagyashreek8) [reply provisioner/ibm-s3fs-provisioner.go:139]: The error we are handling in next line where we pass unix_addr `conn, err := net.DialUnix("unix", nil, unix_addr)` , so for 139 line I will ignore error.

**Review by mssachan — APPROVED** [review]
LGTM

**Review by ambiknai — APPROVED** [review]

**Review by nkkashyap — APPROVED** [review]
lgtm

**Pull request merged**